backend/algorithms/objects/conditions.py

```diff
@@ -364,7 +388,8 @@ def __init__(self, exclusion: str):
         self.exclusion = exclusion
 
     def validate(self, user: User) -> tuple[bool, list[str]]:
-        return not user.in_program(self.exclusion), []
+        excluded = not user.in_program(self.exclusion)
+        return (excluded, []) if excluded else excluded, ["This course cannot be taken in your program"]
```

Comment:
it may be wise to wrap brackets around the part after the else as well because otherwise it *may* come across as `return (excluded .... if ... else ... excluded), ["this course..."]`

Doing this will actually also fix ur `mypy` error on this line because you're returning a bool here as is needed but because of the above thingy, it thinks that the whole thing on the left of the last comma as one entry in the tuple